<?xml version="1.0" ?>
<launch>
	<!-- Arguments for which controller type to utilize -->
	<arg name="control" doc="Which controller type to use" />
	<arg name="full_arm" doc="Control full arm?" />
	<arg name="gripper_enabled" default="false" doc="Enable gripper for description" />
	<arg name="tek4030" default="false" />

	<!-- Load joint controller configurations from YAML file to parameter server -->
	<rosparam if="$(eval control == 'position')"
		file="$(find crustcrawler_hardware)/config/crustcrawler_position.yaml" command="load"/>
	<rosparam if="$(eval control == 'velocity')"
		file="$(find crustcrawler_hardware)/config/crustcrawler_velocity.yaml" command="load"/>
	<rosparam if="$(eval control == 'effort')"
		file="$(find crustcrawler_hardware)/config/crustcrawler_effort.yaml" command="load"/>
	<rosparam if="$(eval control == 'trajectory' and full_arm)"
		file="$(find crustcrawler_hardware)/config/crustcrawler_trajectory.yaml" command="load"/>
	<rosparam if="$(eval control == 'trajectory' and not full_arm)"
		file="$(find crustcrawler_hardware)/config/crustcrawler_trajectory_small.yaml" command="load"/>

	<param name="robot_description"
		command="$(find xacro)/xacro --inorder
		'$(find crustcrawler_description)/urdf/crustcrawler.xacro'
		gripper_enabled:=$(arg gripper_enabled)
		tek4030:=$(arg tek4030)
		full_arm:=$(arg full_arm)" />

	<!-- Load hardware interface node -->
	<node name="crustcrawler_hardware" pkg="crustcrawler_hardware"
		type="hardware_interface" output="screen" ns="/crustcrawler"
		required="true">
		<param name="full_arm" value="$(arg full_arm)" />
	</node>

	<!-- load the control for individual controllers-->
	<node if="$(eval control != 'trajectory' and full_arm)"
		name="controller_spawner" pkg="controller_manager" type="spawner"
		respawn="false" output="screen" ns="/crustcrawler"
		args="joint_state_controller
		joint1_controller
		joint2_controller
		joint3_controller
		joint4_controller
		joint5_controller
		joint6_controller
		"/>
	<node if="$(eval control != 'trajectory' and not full_arm)"
		name="controller_spawner" pkg="controller_manager" type="spawner"
		respawn="false" output="screen" ns="/crustcrawler"
		args="joint_state_controller
		joint1_controller
		joint2_controller
		joint3_controller
		"/>
	<!-- Load control for trajectory following -->
	<node if="$(eval control == 'trajectory')"
		name="controller_spawner" pkg="controller_manager" type="spawner"
		respawn="false" output="screen" ns="/crustcrawler"
		args="joint_state_controller controller">
	</node>

	<!-- Convert joint states to TF transforms for rviz, etc -->
	<node name="crustcrawler_state_publisher" pkg="robot_state_publisher"
		type="robot_state_publisher" respawn="false" output="screen" ns="crustcrawler">
	</node>
</launch>
